cdq-analysis/main.go

```diff
@@ -0,0 +1,79 @@
+//
+// Copyright 2023 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"go.uber.org/zap/zapcore"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	"github.com/redhat-appstudio/application-service/cdq-analysis/pkg"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+func main() {
+	// remove the prefix and suffix quotes
+	for i := 1; i <= 10; i++ {
+		if strings.HasPrefix(os.Args[i], "\"") && strings.HasSuffix(os.Args[i], "\"") && len(os.Args[i]) > 1 {
+			os.Args[i] = os.Args[i][1 : len(os.Args[i])-1]
+		}
+	}
+	gitToken := os.Args[1]
+	namespace := os.Args[2]
+	name := os.Args[3]
+	contextPath := os.Args[4]
+	devfilePath := os.Args[5]
+	dockerfilePath := os.Args[6]
+	URL := os.Args[7]
+	Revision := os.Args[8]
+	DevfileRegistryURL := os.Args[9]
+	isDevfilePresent, _ := strconv.ParseBool(os.Args[10])
+	isDockerfilePresent, _ := strconv.ParseBool(os.Args[11])
+	createK8sJob, _ := strconv.ParseBool(os.Args[12])
+
+	opts := zap.Options{
+		TimeEncoder: zapcore.ISO8601TimeEncoder,
+	}
+	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
+	log := ctrl.Log.WithName("cdq-analysis").WithName("CloneAndAnalyze")
+	var ctx context.Context
+	var clientset *kubernetes.Clientset
+	if createK8sJob {
```

Comment:
Should this flag instead be outside of the module and be part of HAS? Because HAS will be what's launching the Kubernetes Job containing this go module (and main.go)

FWIW, the way I split between using the go module vs. the job for gitops generation was like this: https://github.com/redhat-appstudio/application-service/blob/d24a233c36d89456366207da5a2438498954cb3e/controllers/component_controller.go#L532-L544. Basically if LOCAL_GITOPS was set, or an annotation on the CR was set, then we would invoke the Go module directly, otherwise we would launch the Kubernetes job containing the module

Reply:
Sorry, what I mean is, is that the logic that splits between whether or not to launch a job or to launch a module, should reside in HAS, rather than in the Go module.

Reply:
yes. I understands that, but the go module will need that env as well. since it need to determine if a configmap need to be created when returning the result.  
in this PR, we are using go function to return the result. 
the job not yet created. and will have that ENV been set in HAS when a job is ready to be consumed.